layout/src/main/java/com/vlkan/log4j2/logstash/layout/util/JsonGenerators.java

```diff
@@ -114,4 +115,18 @@ public static void writeObject(JsonGenerator jsonGenerator, Object object) throw
 
     }
 
+    public static void resetGeneratorState(JsonGenerator generator, ByteBuffer buffer) throws IOException {
+        while (true) {
+            JsonStreamContext ctx = generator.getOutputContext();
+            if (ctx.inArray()) {
+                generator.writeEndArray();
+            } else if (ctx.inObject()) {
+                generator.writeEndObject();
+            } else {
```

Comment:
I think `resetGeneratorState` has a short-coming which has already been addressed in `rescueJsonGeneratorState`. Consider the following:

```java
JsonGenerator jsonGenerator = objectMapper.getFactory().createGenerator(outputStream);
jsonGenerator.writeStartObject();
jsonGenerator.writeFieldName("foo");
```

At this state, neither a `]`, nor a `}` would suffice; a `jsonGenerator.writeFieldValue(someRandomValue)` is required. I am working on top of your changes to make this work, FYI.